### DISCUSSION TITLE
React Router v7 Integration

### DISCUSSION THREAD

**Discussion by jeremylynch** [shakacode/shakapacker#528] [General]
React Router v7 has now launched: https://remix.run/blog/react-router-v7

Given that HMR, server rendering are now included in this package, what does this mean for Shakapacker? How can this be integrated into Shakapacker?

**Comment by justin808** [comment]
This would be more relevant for react on rails. 

Cc @Judahmeek

**Comment by Judahmeek** [comment]
React Router v7's framework features are all enabled through a Vite plugin.

[Shakapacker enables HMR, SSR, etc](https://github.com/shakacode/react_on_rails_demo_ssr_hmr), but would require a similar webpack plugin for such features to automatically work without manual setup.
